### ISSUE TITLE
inversion doesn't look like the face of img source

### ISSUE THREAD

**Issue by molo32** [omertov/encoder4editing#2]
inversion doesn't look like the face of img source,How can I make it look more like img source?

**Comment by omertov** [comment]
Hi @molo32,
Can you provide further details? have you performed the required face alignment?

**Comment by molo32** [comment]
```
image_path = "/content/8.jpg"
original_image = Image.open(image_path)
original_image = original_image.convert("RGB")
input_image = run_alignment(image_path)

def run_on_batch(inputs, net):
    images, latents = net(inputs.to("cuda").float(), randomize_noise=False, return_latents=True)
    if experiment_type == 'cars_encode':
        images = images[:, :, 32:224, :]
    return images, latents

def display_alongside_source_image(result_image, source_image):
    res = np.concatenate([np.array(source_image.resize(resize_dims)),
                          np.array(result_image.resize(resize_dims))], axis=1)
    return Image.fromarray(res)

input_image.resize(resize_dims)
img_transforms = EXPERIMENT_ARGS['transform']
transformed_image = img_transforms(input_image)
with torch.no_grad():
    tic = time.time()
    images, latents = run_on_batch(transformed_image.unsqueeze(0), net)
    result_image, latent = images[0], latents[0]
    toc = time.time()
    print('Inference took {:.4f} seconds.'.format(toc - tic))
# Display inversion:
display_alongside_source_image(tensor2im(result_image), input_image)
```


![download](https://user-images.githubusercontent.com/55426197/109432925-4dd54f80-79ec-11eb-9fd7-e5f59ec8753f.png)

**Comment by omertov** [comment]
It seems like you run our encoder correctly.

Generally speaking, our pretrained e4e encoder is specifically designed to balance the tradeoffs existing in the StyleGAN's latent space (See our paper for further details and examples). 
By doing so, we lose some reconstruction accuracy to gain more editable latent codes (that can be better used by other existing latent-space manipulation techniques, StyleFlow for example) compared to other inversion methods. 

If exact reconstruction is what you seek, direct optimization will always yield the best results, or alternatively, you can **control the tradeoff yourself according to your needs**.
For example,  you can train the encoder to favor reconstruction over editablity by not using the latent codes discriminator or by tuning the progressive training parameters.

**Comment by woctezuma** [comment]
Relevant: https://github.com/rolux/stylegan2encoder/issues/2#issuecomment-570191777
